assets/js/socket.ts

```diff
@@ -1,104 +1,37 @@
-// the connection for use in the layout.
-//
-// In your "lib/web/router.ex":
-//
-//     pipeline :browser do
-//       ...
-//       plug MyAuth
-//       plug :put_user_token
-//     end
-//
-//     defp put_user_token(conn, _) do
-//       if current_user = conn.assigns[:current_user] do
-//         token = Phoenix.Token.sign(conn, "user socket", current_user.id)
-//         assign(conn, :user_token, token)
-//       else
-//         conn
-//       end
-//     end
-//
-// Now you need to pass this token to JavaScript. You can do so
-// inside a script tag in "lib/web/templates/layout/app.html.eex":
-//
-//     <script>window.userToken = "<%= assigns[:user_token] %>";</script>
-//
-// You will need to verify the user token in the "connect/2" function
-// in "lib/web/channels/user_socket.ex":
-//
-//     def connect(%{"token" => token}, socket) do
-//       # max_age: 1209600 is equivalent to two weeks in seconds
-//       case Phoenix.Token.verify(socket, "user socket", token, max_age: 1209600) do
-//         {:ok, user_id} ->
-//           {:ok, assign(socket, :user, user_id)}
-//         {:error, reason} ->
-//           :error
-//       end
-//     end
+// Take the query from the URL ( the #bit ) and maps it to an event from the 
+// socket to pheonix
 //
-// Finally, pass the token on connect as below. Or remove it
-// from connect if you don't care about authentication.
-
-socket.connect()
-
-// Now that you are connected, you can join channels with a topic:
-let messagesContainer = document.querySelector("#sidebar ol")
-
-let inquiriesChannel = socket.channel("inquiries", {})
-
-inquiriesChannel.join()
-  .receive("ok", resp => { console.log("Joined successfully", resp) })
-  .receive("error", resp => { console.log("Unable to join", resp) })
-
-inquiriesChannel.on("artworkinquiryrequest.inquired", payload => {
-
-  if (payload.partner_locations.length && payload.user.location) {
-    // Use the furthest away location
-    let partnerLoc = payload.partner_locations[0]
-    payload.partner_locations.forEach(loc => {
-      if (getDistance(payload.user.location, loc) > getDistance(payload.user.location, partnerLoc)) {
-        partnerLoc = loc
-      }
-    });
-
-    addArc(payload.user.location, partnerLoc)
-
-    const distance =  Math.round(getDistance(payload.user.location, partnerLoc))
-
-    let newItem = document.createElement("li", { class: "news-item"})
-    newItem.innerHTML = `
-      <div class="img" style="background-image: url(${payload.artwork.images[0].image_urls.medium});"></div>
-      <p>
-        <span>${payload.properties.inquireable.name}</span> from <span>${payload.partner.name}</span>.<br/>
-        ${shortDateString(payload.user.location)} ✈ ${shortDateString(partnerLoc)} (${distance}km)
-      </p>
-      `
-
-    messagesContainer.insertBefore(newItem, messagesContainer.firstChild);
+const queryToEvent = (query: string) => {
+  switch (query) {
+    case "purchases":
+      return "purchases"
```

Comment:
this will be `purchase.purchased` i _think_ but I'll merge and confirm this locally